portfolio/src/main/webapp/script.js

```diff
@@ -56,3 +56,25 @@ function getRandomName() {
     document.getElementById('name-container').innerText = name;
   });
 }
+
+/**
+ * Fetches names from DataServlet and adds them to the DOM as a list.
+ */
+function getNames() {
+  fetch('/data').then(response => response.json()).then((names) => {
+
+    const namesListElement = document.getElementById('names-container');
+    namesListElement.innerHTML = '';
+    for (const nameIndex in names) {
+      console.log(names[nameIndex]);
```

Comment:
Is this just for debugging purpose, or you intended to print console log when the request returns?
Is the former, please remove it. If the latter, you probably should add some label with the value.